Added --no-confirm option

patillacode opened the pull request:
Added a missing option for experienced users to avoid the confirmation step.
Also some docstrings were missing 😅

inirudebwoy commented:
That is a nice feature to have, as it allows to use this tool in a some other script.

patillacode commented:
Yep... I had it in my TODO list so I am gonna add it now.
Just fixing some docstrings before merging.

inirudebwoy commented on convert/commands/base.py:
```diff
@@ -79,16 +79,18 @@ def get_user_input(self, conversion_data):
+        # multiple files flow
         if self.options['--multiple']:
+            # do not show confirmation message if the option is enabled
+            if not self.options['--no-confirm']:
```
This bit is copied in other branch of this conditional. Have you thought about pulling it out into a separate function?

patillacode replied:
Hmmm, I didn't, it would be kind of the same though...

```
def no_confirm():
    return self.options['--no-confirm']
```

Then in the code you quoted:
```
if not no_confirm():
```

Can you give an example of how you would do it?
I am confused because it is not a piece of code, it is a one liner.

inirudebwoy replied:
Sorry, I did not mean this one line but rather block of code.
```            # do not show confirmation message if the option is enabled
            if not self.options['--no-confirm']:
                if not self.confirm_multiple(source_extension,
                                             source_folder,
                                             conversion_data['extension'],
                                             destination):
                    sys.exit(2)

                # clear screen
                clear()
```

The difference between is a function call or either `confirm_multiple` or `confirm_single`.

```
def confirmator(confirm_func, *args):
            if not self.options['--no-confirm']:
                if not confirm_func(*args):
                    sys.exit(2)

                # clear screen
                clear()
```

This is a really raw pseudo solution but maybe it will illustrate what I have on mind.

patillacode replied:
Oh, yes, I see it.

Thanks, I will give it a go on this PR, cheers!

patillacode commented on convert/converter_utils/utils.py:
```diff
@@ -182,6 +182,13 @@ def clear():
     os.system(command)
 
 
+def user_confirmed(confirmation):
```
@inirudebwoy you gave me the idea for this case while thinking about the one you mentioned. 😄 
Thanks!

patillacode commented:
Ok, so I changed a few things (more than I expected) in this PR:

- Updated README to have latest code changes explained where necessary.
- Options where wrongly defined, I checked `docopt` docs and fixed it.
- Added the `-n` or `--no-confirm` option to avoid the confirmation step (main purpose of the PR)
- After @inirudebwoy comments (thanks again) I refactored some code moving some lines from `base.py` to `utils.py` making the `Base` class more generic, as it should be (for instance the `confirmator` method or the `user_confirmed` method)
- Fixed terminal output (clears, typos, etc)

Missing things (that NEED to exist) for another PR to a void making this one unnecessarily big would be:
- Method's docstrings (review existing ones and write missing ones)
- Testing
- More refactoring maybe?

I might create issues as reminders for these things.

patillacode commented:
This PR fixes #7
